sns_ik_examples/src/ik_tests.cpp

```diff
@@ -209,6 +210,37 @@ void test(ros::NodeHandle& nh, double num_samples_pos, double num_samples_vel, s
   assert(chain.getNrOfJoints() == al.data.size());
 
   // SNS Position Tests
+  struct velocitySolverData {
+    sns_ik::VelocitySolveType type;
+    std::string               name;
+    double                   score;
+    double           scaling_score;
+    double                avg_time;
+  };
+  std::vector<velocitySolverData> vel_solver_data;
+  velocitySolverData sns = {sns_ik::SNS,"SNS",0.0,0.0};
+  vel_solver_data.push_back(sns);
+  velocitySolverData sns_optimalsm = {sns_ik::SNS_OptimalScaleMargin,"SNS Optimal Scale Margin",0.0,0.0};
+  vel_solver_data.push_back(sns_optimalsm);
+  velocitySolverData sns_optimal = {sns_ik::SNS_Optimal,"SNS Optimal",0.0,0.0};
+  vel_solver_data.push_back(sns_optimal);
+  velocitySolverData sns_fast = {sns_ik::SNS_Fast,"SNS Fast",0.0,0.0};
+  vel_solver_data.push_back(sns_fast);
+  velocitySolverData sns_fastoptimal = {sns_ik::SNS_FastOptimal,"SNS Fast Optimal",0.0,0.0};
+  vel_solver_data.push_back(sns_fastoptimal);
+  double linearTolerance = 0.001;
+  double angularTolerance = 0.001;
```

Comment:
The tolerances should be kept at 1e-5 to match the tolerances used by kdl and trac-ik.